### PR TITLE
Use unique uninstall & doc targets

### PR DESCRIPTION
**Pull request by powderluv**
When Pangolin is used in other projects these targets collide

Also fixes https://github.com/stevenlovegrove/Pangolin/issues/355 

TEST: Build on OSX and run samples

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -640,7 +640,7 @@ find_package(Doxygen)
 if(DOXYGEN_FOUND)
     configure_file(${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile.in ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile @ONLY)
     file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../doc )
-    add_custom_target(doc
+    add_custom_target(pangolin_doc
```
Comment (NikolausDemmel) [inline src/CMakeLists.txt]: You should also update the docstring above that currently reads `(make doc)`

**Comment by NikolausDemmel** [comment]
Thanks for this PR. I think these changes should be merged right away.

> Also fixes #355

However, to address all points in #355 one would also have to resolve the issue when using this with a specified Eigen version that is internal to the parent project (i.e. within the source directory of the parent project), so I don't think this PR should close #355.

File: src/CMakeLists.txt (hunk 1 above)
Comment (powderluv) [reply src/CMakeLists.txt]: done

**Comment by powderluv** [comment]
Ok on not closing #355. I have a few more patches to help build on windows / with GLEW variations etc  that I will clean up from https://github.com/powderluv/Pangolin and submit PRs.

**Comment by NikolausDemmel** [comment]
Awesome thanks.

:+1: for merging.

**Comment by stevenlovegrove** [comment]
Thanks!

**Comment by leCaTo** [comment]
An updated README with the new target name would be nice:
'cmake --build . --target pangolin_doc' instead of 'cmake --build . --target doc'

**Comment by NikolausDemmel** [comment]
@leCaTo can you send a PR?

**Pull request merged**